src/h_matrix.hpp

```diff
@@ -551,8 +551,14 @@ template<typename T> class HMatrix : public Tree<HMatrix<T> >, public RecursionM
   HMatrix<T>* get(int i, int j) const {
     assert(i>=0 && i<nrChildRow());
     assert(j>=0 && j<nrChildCol());
-    assert(i + j * nrChildRow() < this->nrChild());
-    return this->getChild(i + j * nrChildRow());
+    if ( this->nrChild() ) {
+	assert(i + j * nrChildRow() < this->nrChild());
+	return this->getChild(i + j * nrChildRow());
+    }
+    else {
+	assert(i + j * nrChildRow() == 0);
+	return (HMatrix<T>*)this;
+    }
```

Comment:
In which contest do you need this ? can't you do the test in the caller ?

Reply:
I need it if the HMatrix has only one level. Thus you can only get the HMatrix itself as it has no children.

Reply:
Its better to remove this and do the test in the caller (or change the way your cluster tree is created to be sure to have at least 1 child at the root)

Reply:
Done

Reply:
Done